src/test/java/org/jabref/logic/importer/ParserResultTest.java

```diff
@@ -48,4 +53,109 @@ public void doesNotHaveDuplicateWarnings() {
         parserResult.addWarning("Duplicate Warning");
         assertEquals("Duplicate Warning", parserResult.getErrorMessage());
     }
+
+    @Test
+    public void warningAddedForWhitespaceInCitationKeyImport() throws IOException {
+        //whitespace after citation key "myArticle "
+        String bibtexEntry = """
+                @article{ myArticle ,
+                  author    = "Author Name",
+                  title     = "Title of the Article",
+                  journal   = "Journal Name",
+                  year      = "2024",
+                  pages     = "1-10",
+                  publisher = "Publisher Name"
+                }
+                """;
+
+        File tempFile = File.createTempFile("invalidBibTex", ".bib");
+        try {
+            Files.write(tempFile.toPath(), bibtexEntry.getBytes());
+            ParserResult parserResult = OpenDatabase.loadDatabase(tempFile.toPath(), mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS), new DummyFileUpdateMonitor());
+            assertEquals(parserResult.warnings().getFirst(),"Line 1: Found corrupted citation key  (contains whitespaces).");
+        } finally {
+            if (!tempFile.delete()) {
```

Comment:
No need to delete temp files. This is handled by JUnit. Please use `@TempDir` annotation to make use of that functionality.